chainbase/src/main/java/org/tron/core/db/common/iterator/DBIterator.java

```diff
@@ -1,9 +1,23 @@
 package org.tron.core.db.common.iterator;
 
+import com.google.common.collect.Iterators;
+import com.google.common.collect.UnmodifiableIterator;
+import com.google.common.primitives.Bytes;
 import java.io.Closeable;
 import java.util.Iterator;
 import java.util.Map.Entry;
 
-public interface DBIterator extends Iterator<Entry<byte[], byte[]>>, Closeable {
+public interface DBIterator extends Iterator<Entry<byte[], byte[]>>, AutoCloseable, Closeable {
 
+  void seek(byte[] key);
+
+  void seekToFirst();
+
+  void seekToLast();
+
+  default UnmodifiableIterator<Entry<byte[], byte[]>> prefixQueryAfterThat
+      (byte[] key, byte[] afterThat) {
+    this.seek(afterThat == null ? key : afterThat);
+    return Iterators.filter(this, entry -> Bytes.indexOf(entry.getKey(), key) == 0);

```

Comment:
Here I have doubt, whether this iterator will still access all data in the database but just return the data that matched the filter.
If so, it will do a lot of useless read-op?

Reply:
Not All, iterates over all data after the prefix. I can't think of a better way for now. It takes 2 minutes to iterate the database.

Reply:
Use the default iterator and move the filter logic into `startRewardCal`, or re-implement a prefix iterator.
Seems 2 minutes is not a big deal, depends on you